backend/tilavarauspalvelu/models/application_round/model.py

```diff
@@ -94,6 +94,38 @@ class Meta:
         verbose_name = _("application round")
         verbose_name_plural = _("application rounds")
         ordering = ["pk"]
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(application_period_begin__lt=models.F("application_period_end")),
+                name="application_period_begin_before_end",
+                violation_error_message="Application period must begin before it ends",
+            ),
+            models.CheckConstraint(
+                check=models.Q(reservation_period_begin__lt=models.F("reservation_period_end")),
+                name="reservation_period_begin_before_end",
+                violation_error_message="Reservation period must begin before it ends",
+            ),
+            models.CheckConstraint(
+                check=models.Q(public_display_begin__lt=models.F("public_display_end")),
+                name="public_display_begin_before_end",
+                violation_error_message="Public display period must begin before it ends",
+            ),
+            models.CheckConstraint(
+                check=models.Q(application_period_end__date__lt=models.F("reservation_period_begin")),
+                name="applications_end_before_reservation_period_begins",
+                violation_error_message="Application period must end before reservation period begins",
+            ),
+            models.CheckConstraint(
+                check=~models.Q(handled_date__isnull=True, sent_date__isnull=False),
+                name="must_handle_before_sending",
+                violation_error_message="Application round must be handled before it can be sent",
+            ),
+            models.CheckConstraint(
+                check=models.Q(handled_date=None) | models.Q(application_period_end__date__lt=models.F("handled_date")),
+                name="handling_after_application_period_end",
+                violation_error_message="Application round can only be handled after its application round has ended",
+            ),
```

Comment:
Would adding a constraint for "`handled_date` must not be after `sent_date`" be useful also 🤔 
That would also cover the case of "`sent_date` must be after `application_period_end`"